core/src/main/java/com/exadel/kaliada/core/servlets/LikeComponentServlet.java

```diff
@@ -0,0 +1,84 @@
+ *  limitations under the License.
+ */
+package com.exadel.kaliada.core.servlets;
+
+import org.apache.jackrabbit.commons.JcrUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.servlets.annotations.SlingServletPaths;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.propertytypes.ServiceDescription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.*;
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.lang.module.ResolutionException;
+
+/**
+ * Servlet that writes some sample content into the response. It is mounted for
+ * all resources of a specific Sling resource type. The
+ * {@link SlingSafeMethodsServlet} shall be used for HTTP methods that are
+ * idempotent. For write operations use the {@link SlingAllMethodsServlet}.
+ */
+@Component(service = { Servlet.class })
+@SlingServletPaths(value = "/bin/like")
+@ServiceDescription("Like component Servlet")
+public class LikeComponentServlet extends SlingAllMethodsServlet {
+
+    private static final long serialVersionUID = 1L;
+    private static final Logger logger = LoggerFactory.getLogger(LikeComponentServlet.class);
+    private static final String NODE_NAME = "/content/aemdemo/us/en/component-basics/jcr:content/root/container/container/like";
+    private static final long START_COUNT = 0;
+
+    @Override
+    protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response) throws ServletException, IOException {
+        super.doGet(request, response);
+    }
+
+    @Override
+    protected void doPost(final SlingHttpServletRequest req,
+            final SlingHttpServletResponse resp) throws ServletException, IOException {
+        try{
+            Resource resource = req.getResource();
+            ResourceResolver resourceResolver = resource.getResourceResolver();
+            Session session = resourceResolver.adaptTo(Session.class);
+            Node node = session.getNode(NODE_NAME);
+            String propertyName = req.getParameterNames().nextElement();
+            if (!node.hasProperty(propertyName)){
+                node.setProperty(propertyName, START_COUNT);
+            }
+            long likes = node.getProperty(propertyName).getValue().getLong();
+            likes = changeLikeValue(likes, req.getParameter(propertyName));
+            node.getProperty(propertyName).setValue(likes);
+            session.save();
+            resp.setContentType("text/html");
+            resp.getWriter().write(String.valueOf(likes));
+        }catch (RepositoryException e){
+            logger.error(e.getMessage());
```

Comment:
It's better to explain in error message, what happened and when (in a human-like language, and then attach the exception as is). The message is for "human" explanation, while the exception is for stack trace.
Usually it goes like this:
```
LOG.error("Could not retrieve value request was {}", req.getPath(), e)
```
The logger methods are variadic, you can specify any number of params, they will be inserted into the initial string at the `{}` placeholders. Last argument is the exception object. If you specify it, you get the stacktrace. If you do with something like e.getMessage(), you lose the stacktrace;)